src/password.cpp

```diff
@@ -0,0 +1,45 @@
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <wordexp.h>
+
+#include "password.h"
+
+const std::string read_password(const char* file) {
+    wordexp_t exp;
+    std::ifstream ifs;
+    std::string pass = "";
+
+    wordexp(file,&exp,0);
+    if (exp.we_wordc > 0) {
+        ifs.open(exp.we_wordv[0]);
+    }
+    wordfree(&exp);
+    if (!ifs.is_open()) {
+    } else {
+        std::getline(ifs, pass);
+    }
+    return pass;
+}
+
+const std::string eval_password(const char* cmd) {
+    char buffer[128];
+    std::string pass = "";
+
+    FILE* pipe = popen(cmd, "r");
+    if (!pipe) {
+    } else {
+        try {
+            while (!feof(pipe)) {
+                if (fgets(buffer, 128, pipe) != NULL) {
+                    buffer[strcspn(buffer, "\r\n")] = 0;
```

Comment:
> If I recall correctly inserting a '\0' into the buffer (in this case replacing '\r' or '\n', whichever comes first, with '\0') should result in "terminating" the string and ignoring everything afterwards.

That's right, but consider what `fgets` returns:

1. If he input line was longer than 127 characters (not a typo—it's 127), `buffer` won't contain any newlines, `strcspn` will return 127, terminating zero will be overwritten with zero (so no harm done), then `buffer` is appended to `pass`. The code then proceeds to new iteration to read more input. This is fine and indeed a sane behaviour.

2. If the input line was shorter than 127 characters, `fgets` will return it *with the newline at the end*. The newline is then stripped, and `buffer` is appended to `pass`. The code then proceeds to new iteration to read more input—i.e. it'll read more lines, which isn't what you intended.

Here's how I tested this:

1. I put the following into `script` next to Newsbeuter:

    ```shell
    #!/bin/sh

    echo aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
    echo bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
    echo cccccccccccccccccccccccccccccccccccccccc
    echo 4444
    echo hello
    echo world
    ```

2. I replaced the body of `main` in `newsbeuter.cpp` with the following:

    ```c++
	auto password = eval_password("./script");
	std::cerr << password << std::endl;
	return 1;
    ```

    (this isn't the cleanest way to do it, but it's quick and easy).

3. Re-compiled and run `./newsbeuter`. It printed out `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbcccccccccccccccccccccccccccccccccccccccc4444helloworld`, i.e. all the input lines concatenated into one.

The way I'd go about this is:

1. use `utils::get_command_output` to get the output;
2. use something along the lines of `std::erase(pass.find_first_of("\r\n"), pass.end())` to extract the first line.